src/insights_client/__init__.py

```diff
@@ -351,37 +349,62 @@ def _main():
     attempt to collect and upload with new, then current, then rpm
     if an egg fails a phase never try it again
     """
+    set_up_logging()
+    logger.debug("Verifying initial eggs to load config.")
+
+    validated_eggs = []
+    for egg in [STABLE_EGG, RPM_EGG]:
+        logger.debug("GPG verifying initial egg %s..." % (egg,))
+        if gpg_validate(egg):
+            logger.debug("GPG verification successful. Loading initial egg: %s" % (egg,))
+            validated_eggs.append(egg)
+        else:
+            logger.debug("WARNING: GPG verification failed. Not loading initial egg: %s" % (egg,))
     # sort rpm and stable eggs after verification
-    validated_eggs = sorted_eggs(list(filter(gpg_validate, [STABLE_EGG, RPM_EGG])))
+    validated_eggs = sorted_eggs(validated_eggs)
     # if ENV_EGG was specified and it's valid, add that to front of sys.path
     #  so it can be loaded initially. keep it in its own var so we don't
     #  pass it to run_phase where we load it again
+    logger.debug("Verifying initial ENV_EGG: %s..." % (ENV_EGG,))
     if gpg_validate(ENV_EGG):
+        logger.debug("GPG verification successful. Loading initial egg: %s" % (ENV_EGG,))
         valid_env_egg = [ENV_EGG]
     else:
+        logger.debug("WARNING: GPG verification failed. Not loading initial egg: %s" % (ENV_EGG,))
         valid_env_egg = []
 
     if not validated_eggs and not valid_env_egg:
-        sys.exit("No GPG-verified eggs can be found")
+        sys.exit("No GPG-verified initial eggs can be found")
 
     # ENV egg comes first
     sys.path = valid_env_egg + validated_eggs + sys.path
 
     try:
+        sys_path = ":".join(sys.path)
+        logger.debug("Importing the insights module from %s..." % (sys_path,))
```

Comment:
I have my doubts about the benefit of logging full PATH, but I am not against either.

Non-blocking comment.

Reply:
It is maybe a bit verbose. But when debugging why was a module loaded from a certain location, the full PATH may be invaluable.